### PR TITLE
Storage QoS recipes

### PR DESCRIPTION
**Pull request by artek-koltun**

### CODE REVIEW

File: build/storage/recipes/README.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ fio traffic by means of `host-target` container through a hot-plugged virtio-blk
 device.
 * [scale-out](./virtio-blk/scale-out.md) - similar to hot-plug recipe but describing
 how to hot-plug/hot-unplug 64 virtio-blk devices.
+* [qos](./virtio-blk/qos.md) - in the recipe, it is explained how to set QoS
```
[the remark this answers is not in the document]
Comment (artek-koltun) [reply build/storage/recipes/README.md]: changed

[the remark this answers is not in the document]
Comment (artek-koltun) [reply build/storage/recipes/README.md]: done

File: build/storage/recipes/nvme/qos.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+1. Perform all steps described in [environment setup](../environment_setup.md)
+
+2. Run [fio scenario](fio.md) to observe device throughput.
+In the fio command `rw` field can be replaced with `randread` or `randwrite` to exercise device only with reads or writes.
+
+3. Set required QoS limits.
+IPDK Storage Solution leverages SPDK SMA to get supported [QoS capabilities](https://github.com/spdk/spdk/blob/aed4ece93c659195d4b56399a181f41e00a7a25e/python/spdk/sma/proto/sma.proto#L148) and set [required QoS limits](https://github.com/spdk/spdk/blob/aed4ece93c659195d4b56399a181f41e00a7a25e/python/spdk/sma/proto/sma.proto#L124).
+
+To request supported nvme capabilities, execute the following cmd from `cmd-sender`:
```
[the remark this answers is not in the document]
Comment (artek-koltun) [reply build/storage/recipes/nvme/qos.md]: fixed

File: build/storage/recipes/nvme/qos.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+$ set_max_qos_limits <ipu_storage_container_platform_ip> 8080 $nvme0 $malloc0 0 0 0 16 0 0
+```
+
+Unfortunately, KVM case NVMe devices do not support per-device QoS limits.
+If there is an IPU providing such capabilities, per-device limits can be set by leaving `<volume_id>` argument empty.
+```
+$ set_max_qos_limits <ipu_storage_container_platform_ip> 8080 $nvme0 "" 0 0 0 16 0 0 8080
+```
+
+4. Check if limit was applied
```
[the remark this answers is not in the document]
Comment (artek-koltun) [reply build/storage/recipes/nvme/qos.md]: changed

**Review by intelfisz — APPROVED** [review]

**Pull request merged**